src/diffusion/model.py

```diff
@@ -6,22 +6,28 @@
 from torch import nn
 from transformers import AutoModel
 from transformers import BertConfig
+from transformers import T5EncoderModel
 
 from src.diffusion.utils import configure_schedule, get_x0, get_diffusion_variables, scale_input
-from src.sampling import sample
 from src.metrics import calculate_batch_ce
+from src.sampling import sample
 from src.utils import zero_rank_info
 
 
 def get_components(name: str, mode: str, **model_kwargs):
-    model = AutoModel.from_pretrained(name)
-
     if mode == "same":
+        model = AutoModel.from_pretrained(name)
         return model.encoder, model.decoder, model.config.d_model
 
     elif mode == "bert":
```

Comment:
Let's use options: `blenderbot` and `t5`. In each option we would load encoder and initialize decoder from BERT.